### PR TITLE
support for formatting in fields

### PR DESCRIPTION
**Pull request by vdua**
## depends on 

#4 

## Test URLs:
- Before: https://form-block2--nedbank--vdua.hlx.live/en/personal/borrow/personal-loans-form
- After: https://form-block3--nedbank--vdua.hlx.live/en/personal/borrow/personal-loans-form

## Update
The review comments have been merged in #1

### CODE REVIEW

File: blocks/form/formatting.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+function toString(num) {
+  const [integer, fraction] = num.toString().split('.');
+  const formattedInteger = integer.replace(/(\d)(?=(\d\d\d)+(?!\d))/g, `$1${groupingSymbol}`);
+  let formattedFraction = fraction || minFractionDigits;
+  const lengthFraction = formattedFraction.length;
+  if (lengthFraction < minFractionDigits) {
+    formattedFraction += Array(minFractionDigits - lengthFraction).fill(0).join('');
+  }
+  return `${formattedInteger}${decimalSymbol}${fraction || '00'}`;
```
Comment (vdua) [inline blocks/form/formatting.js:13]: This code is present in nedbank's website right now

File: blocks/form/form.js [hunk 2]
```diff
@@ -84,7 +86,12 @@ function createOutput(fd) {
   const wrapper = createFieldWrapper(fd);
   const output = document.createElement('output');
   output.name = fd.Name;
-  output.innerText = fd.Value;
+  const displayFormat = fd['Display Format'];
+  if (displayFormat) {
+    output.dataset.displayFormat = displayFormat;
+  }
+  const formatFn = formatFns[displayFormat] || formatFns.identity;
```
Comment (trieloff) [inline blocks/form/form.js:93]:
```suggestion
  const formatFn = formatFns[displayFormat] || (x) => x;
```

this makes it a bit more robust against the identity formatter missing.

File: blocks/form/form.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import formatFns from './formatting.js';
```
Comment (trieloff) [inline blocks/form/form.js:1]: good idea to extract the formatter functions. To make this a bit more robust, I'd try to make the import `async` using `import()` and have a fallback.

File: blocks/form/form.js (hunk 2 above)
Comment (vdua) [reply blocks/form/form.js:93]: Resolved.
https://github.com/vdua/nedbank/pull/1/files#diff-14d0faad899dcfcb4e5cd318658f4d58b9e2aa66ebc5912bbe09302339f6a564R156

File: blocks/form/form.js (hunk 3 above)
Comment (vdua) [reply blocks/form/form.js:1]: resolved https://github.com/vdua/nedbank/pull/1/files#diff-14d0faad899dcfcb4e5cd318658f4d58b9e2aa66ebc5912bbe09302339f6a564R1

**Pull request closed without merging**